resolver.go

```diff
@@ -81,6 +102,10 @@ func (r *Resolver) LookupIPAddr(ctx context.Context, domain string) (result []ne
 		}
 	}
 
+	if r.maxCacheTTL != NoMaxCacheTTL {
```

Comment:
Given that we add entries to the cache even if `ttl == 0`, can we just `maxCacheTTL == 0` to mean that we're not applying any max TTL?